drivers/hmis/app/graphql/mutations/update_external_form_submission.rb

```diff
@@ -8,24 +8,71 @@ module Mutations
   class UpdateExternalFormSubmission < CleanBaseMutation
     argument :input, Types::HmisSchema::ExternalFormSubmissionInput, required: true
     argument :id, ID, required: true
+    argument :project_id, ID, required: false # not required for backwards compatibility
 
-    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: false
+    field :external_form_submission, Types::HmisSchema::ExternalFormSubmission, null: true
     field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
 
-    def resolve(id:, input:)
+    def resolve(...)
+      Hmis::Hud::Base.transaction do
+        _resolve(...)
+      end
+    end
+
+    protected
+
+    def _resolve(id:, project_id:, input:)
       record = HmisExternalApis::ExternalForms::FormSubmission.find(id)
       raise 'Access denied' unless allowed?(permissions: [:can_manage_external_form_submissions])
 
       record.assign_attributes(**input.to_params)
 
-      errors = []
-      if record.valid?
-        record.save!
-      else
-        errors = record.errors
-        record = nil
+      errors = HmisErrors::Errors.new
+
+      unless record.valid?
+        errors.add_ar_errors(record.errors&.errors)
+        return { errors: errors }
       end
 
+      if record.status_changed? && record.status == 'reviewed' && !record.spam
+        definition = record.definition
+
+        # Only if there are Client and/or Enrollment fields in the form definition, initialize an enrollment
+        # (which will in turn initialize a Client, inside the form processor).
+        if definition.link_id_item_hash.values.find { |item| ['ENROLLMENT', 'CLIENT'].include?(item.mapping.record_type) }
+          project = Hmis::Hud::Project.find(project_id)
+          record.build_enrollment(project: project, data_source: project.data_source, entry_date: record.created_at)
+        end
+
+        form_processor = record.form_processor || record.build_form_processor(definition: definition)
+
+        form_processor.values = record.cleaned_values
```

Comment:
This part is a bit funny - at the least I'll add comments to try to explain what is funny about it, but perhaps we want to rework it a bit.

`record.cleaned_values` is the raw data from the form submission minus any irrelevant keys that aren't present on the form definition. It's created [here](https://github.com/greenriver/hmis-warehouse/pull/4657/files#diff-34c13c8335974559751928b36cbca0a3363274e0103ed5062da19bf038a918a7R44-R46) and saved to a new column on the external form submission record. e.g. `{"first_name": "martha"}`

`input.hud_values` is transformed like `{"Client.firstName": "martha"}` and I have it coming from the frontend [here](https://github.com/greenriver/hmis-frontend/blob/fccfab5cc2e5ac955ffee60beb31632e1724684d/src/modules/form/util/formUtil.ts#L1306-L1315), because I didn't want to duplicate that logic from ruby code, but I do think it's a little odd that the frontend has to pass this at the review step since there's nothing inherently about it that comes from the user interaction at this stage. Let's discuss more...

Reply:
Yeah, I think it would be good to move that transformation logic to the backend eventually, but probably as a separate chunk of work.

With the change from https://github.com/greenriver/hmis-warehouse/pull/4654, the submitted raw values should have the same format as "hud_values". There is no "values", though, which might cause a problem?

Reply:
decision: remove hud_values from schema, get it from the record's raw_values; remove cleaned_values from db (just use raw_values), don't call validation step that normally uses values, since the form is already submitted